### PR TITLE
[Storage] Fix errors impacting core ci pipeline

### PR DESCRIPTION
**Pull request by m-redding**
There were some storage breaking changes (previously optional parameters are no longer optional) introduced in https://github.com/Azure/azure-sdk-for-net/pull/45336 which were causing the net - core - ci pipeline to fail. There was also a package reference that should have been a project reference that was causing build failures in Azure.Storage.sln.

### CODE REVIEW

**Comment by m-redding** [comment]
The timeout errors are unrelated and are happening on all storage PRs right now  (e.g. https://github.com/Azure/azure-sdk-for-net/pull/45352) so will need to override. This PR is only trying to fix the core ci, not the storage ci, and unblock upgrading the Azure.Core version

File: sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/Azure.Storage.DataMovement.Files.Shares.csproj
Comment (jsquire) [inline sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/Azure.Storage.DataMovement.Files.Shares.csproj:23]: We should probably create an issue that tracks changing this back before release.   But I'll defer to @seanmcc-msft and team.

//cc: @amnguye, @schaabs

**Review by jsquire — APPROVED** [review]

File: sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/Azure.Storage.DataMovement.Files.Shares.csproj
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/Azure.Storage.DataMovement.Files.Shares.csproj:23]: I'm fine with this change right now.

Actually I'm actually working on fixing this right now. I'm actually trying to make a bigger change that way we don't have to keep swapping back and forth as often.

This is mostly here because our tests have a dependency on whenever we update our `ServiceVersion` and the main branch is in development.

**Review by amnguye — APPROVED** [review]
LGTM - Storage DataMovement

File: sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/Azure.Storage.DataMovement.Files.Shares.csproj
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/Azure.Storage.DataMovement.Files.Shares.csproj:23]: Agreed, we actually need this change before we branch for STG 95.

**Review by seanmcc-msft — APPROVED** [review]
We should run storage and dm lib CI on this PR before merging.

**Pull request merged**
